### PR TITLE
Ports: Add GNU gzip utility port

### PR DESCRIPTION
**Pull request by supercomputer7**

### CODE REVIEW

File: Ports/gzip/package.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=gzip
+version=1.12
+useconfigure="true"
+use_fresh_config_sub="true"
```
Comment (timschumi) [inline Ports/gzip/package.sh:5]:
```suggestion
port='gzip'
version='1.12'
useconfigure='true'
use_fresh_config_sub='true'
```

File: Ports/gzip/package.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+workdir="gzip-${version}"
```
Comment (timschumi) [inline Ports/gzip/package.sh:6]: This is the default value for `workdir` and can therefore be omitted.

File: Ports/gzip/package.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+config_sub_paths=("build-aux/config.sub")
+files="https://ftp.gnu.org/gnu/gzip/gzip-${version}.tar.gz gzip-${version}.tar.gz
+https://ftp.gnu.org/gnu/gzip/gzip-${version}.tar.gz.sig gzip-${version}.tar.gz.sig
+https://ftpmirror.gnu.org/gnu/gnu-keyring.gpg gnu-keyring.gpg"
+auth_type="sig"
```
Comment (timschumi) [inline Ports/gzip/package.sh:11]:
```suggestion
auth_type='sha256'
```

Signature-based authentication is a bit flawed, especially if you aren't checking the keyring that you are downloading from the server. I'd strongly prefer using normal sha256sums for checking downloads.

**Review by timschumi — CHANGES_REQUESTED** [review]
Just out of curiosity, is there anything in particular that our version of `gzip` is lacking? :^)

**Comment by supercomputer7** [comment]
> Just out of curiosity, is there anything in particular that our version of `gzip` is lacking? :^)

Well, like how you ported coreutils in the manner of using it to compare performance and such, I did the same for the same reason. Also, by looking into each utility help menu, it seems that we indeed lack some nice features in our implementation.
Also, it's just nice to have the option to use something else on Serenity. I like how we write our utilities ourselves, but sometimes I rather use what GNU offers, just because I'm used to it. Hope this makes some sense :)

**Comment by timschumi** [comment]
> Hope this makes some sense :)

It does, I just wanted to check if this is a port that was made out of necessity or if it was a "this would be nice to have" situation. :^)

**Review by timschumi — APPROVED** [review]
LGTM!

**Pull request merged**